apps/mochi-web/components/settings/general/GeneralPage.tsx

```diff
@@ -1,24 +1,58 @@
-import { Button, useLoginWidget } from '@mochi-ui/core'
+import { Button, Typography, useLoginWidget } from '@mochi-ui/core'
 import { useRouter } from 'next/router'
 import React from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
 import { ROUTES } from '~constants/routes'
+import { GeneralFormValue } from './types'
+import { MoneySource } from './MoneySource'
+import { ReceiverPlatform } from './ReceiverPlatform'
+import { TokenPriority } from './TokenPriority'
 
 export const GeneralPage = () => {
+  const form = useForm<GeneralFormValue>({
+    mode: 'all',
+    defaultValues: {
+      defaultMoneySource: 'mochi',
+      defaultReceiverPlatform: 'discord',
+      defaultTokenPriority: [{ id: '941f0fb1-00da-49dc-a538-5e81fc874cb4' }],
+    },
+  })
+  const { handleSubmit, control, watch } = form
+
   const { logout } = useLoginWidget()
   const { replace } = useRouter()
 
+  const onSubmit = (data: any) => {
+    alert(JSON.stringify(data))
+  }
```

Comment:
Make sure we have the UI for saving data like the application detail page

<img width="994" alt="image" src="https://github.com/consolelabs/mochi-ui/assets/12707960/44b106f6-87a3-4bf1-95d3-74e8abdc6264">

Reply:
And I think we can make the component to be a common one

Reply:
I will add the action bar later